### PR TITLE
[rfc] bigquery observable source asset factory

### PR DESCRIPTION
**Pull request by dpeng817**
Initial design concept for a bigquery observable source asset, which tracks the freshness of a provided table. The purpose of this PR is to gather feedback on the method used for retrieval of freshness from bigquery. If we're in agreement here, I'll transform the retrieval step into a method on the bigquery resource

I don't expect this to land for two reasons. One, this information should live in metadata, not the dataversion specifier. Two, any factory should be able to ingest multiple tables at once as opposed to just one.

How I tested this
This PR mainly exists as a proof of concept of the retrieval method of source freshness. To that end, I've added a test which runs against a live ephemeral bigquery table, and retrieves it's freshness.

### CODE REVIEW

**Comment by dpeng817** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19691** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19691?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19691?utm_source=stack-comment).

File: python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/assets.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+from dagster import DataVersion, observable_source_asset
+
+from .resources import BigQueryResource
+
+"""
+Running the following query in bigquery gets you the last updated timestamp:
```
Comment (krukowski) [inline python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/assets.py:7]: Do we know what timezone this returns in? Can we safely assume it's always UTC?

Comment (dpeng817) [reply python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/assets.py:7]: the field we're operating on is already an int timestamp, timestamp_millis just converts to a utc date translateable timestamp. See here: https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timestamp_millis

**Comment by dpeng817** [comment]
Going to move forward with this approach as a resource method.

**Pull request closed without merging**